arbitrum-client/src/client/mod.rs

```diff
@@ -36,56 +43,31 @@ pub struct ArbitrumClientConfig {
     pub chain: Chain,
     /// HTTP-addressable RPC endpoint for the client to connect to
     pub rpc_url: String,
-    /// The private keys of the accounts to use for signing transactions.
-    /// Multiple keys can be provided to mitigate nonce contention across a node
-    /// / cluster.
-    pub arb_priv_keys: Vec<LocalWallet>,
+    /// The private key of the account to use for signing transactions
+    pub private_key: PrivateKeySigner,
     /// The interval at which to poll for event filters and pending transactions
     pub block_polling_interval_ms: u64,
 }
 
-/// A type alias for the RPC client, which is an ethers middleware stack that
-/// includes a signer derived from a raw private key, and a provider that
-/// connects to the RPC endpoint over HTTP.
-pub type MiddlewareStack =
-    NonceManagerMiddleware<SignerMiddleware<Provider<Http>, Wallet<SigningKey>>>;
-
 impl ArbitrumClientConfig {
     /// Gets the block number at which the darkpool was deployed
     fn get_deploy_block(&self) -> BlockNumber {
         match self.chain {
-            Chain::Mainnet => BlockNumber::Number(MAINNET_DEPLOY_BLOCK.into()),
-            Chain::Testnet => BlockNumber::Number(TESTNET_DEPLOY_BLOCK.into()),
-            Chain::Devnet => BlockNumber::Number(DEVNET_DEPLOY_BLOCK.into()),
+            Chain::Mainnet => MAINNET_DEPLOY_BLOCK,
+            Chain::Testnet => TESTNET_DEPLOY_BLOCK,
+            Chain::Devnet => DEVNET_DEPLOY_BLOCK,
         }
     }
 
     /// Constructs RPC clients capable of signing transactions from the
     /// configuration
-    async fn get_accounts(&self) -> Result<Vec<Arc<MiddlewareStack>>, ArbitrumClientConfigError> {
-        let mut provider = Provider::<Http>::try_from(&self.rpc_url)
-            .map_err(|e| ArbitrumClientConfigError::RpcClientInitialization(e.to_string()))?;
-
-        provider.set_interval(Duration::from_millis(self.block_polling_interval_ms));
-
-        let chain_id = provider
-            .get_chainid()
-            .await
-            .map_err(|e| ArbitrumClientConfigError::RpcClientInitialization(e.to_string()))?
-            .as_u64();
-
-        // Build the RPC clients
-        let accounts = self
-            .arb_priv_keys
-            .iter()
-            .map(|wallet| {
-                let account = wallet.clone().with_chain_id(chain_id);
-                let addr = account.address();
-                Arc::new(provider.clone().with_signer(account).nonce_manager(addr))
-            })
-            .collect();
-
-        Ok(accounts)
+    fn get_provider(&self) -> Result<RenegadeProvider, ArbitrumClientConfigError> {
+        let url = Url::parse(&self.rpc_url)
+            .map_err(err_str!(ArbitrumClientConfigError::RpcClientInitialization))?;
+        let key = self.private_key.clone();
+        let provider = ProviderBuilder::new().wallet(key).on_http(url);
```

Comment:
just confirming, this should have the nonce management & chain id fillers by default, right?